### PR TITLE
deps: remove sl-blip

### PR DESCRIPTION
**Pull request by rmg**
Replace with a custom script that better handles restricted networks by enforcing tight timeouts.

Another alternative to the CDN based approach is to use Google Analytics, as in https://github.com/mktmpio/node-mktmpio/blob/master/scripts/post.js

@sumitha @sam-github @ijroth 

Connect to strongloop-internal/scrum-nodeops#1028

### CODE REVIEW

**Comment by sumitha** [comment]
I'm okay with any approach if we can collect the data, to get the IP address

**Comment by rmg** [comment]
It basically does the same thing as the dependency version except for 2 differences:
 * will block the install for at most 1 second (dependency version uses a much longer timeout)
 * will **not** record installations from a git checkout (will make our numbers more accurate but likely much lower because of removing CI builds)

**Comment by raymondfeng** [comment]
I don't like the idea to have a `postinstall` script at the top level. The script will be skipped if the current user is `root`. What about the following:

1. Create a regular module such as `strongloop-analytics` and make it an optional dep of our modules
2. Put the postinstall script to ` strongloop-analytics`

**Comment by rmg** [comment]
@raymondfeng this is just an example module. The same script would be added to the dependencies as well.

Another side effect with using a separate module is it would only get installed and counted once, at the top level, since all the dependencies that also depend on it would no longer install it. That _could_ be a good thing..

**Comment by sam-github** [comment]
postinstall isn't skipped by root, to my knowledge.

I dislike the idea of dumping this code in every module... I'd like it much better if it was something that slt-release did before publish, so that it wouldn't be comitted to git, and could be maintained in one place.

Is blip really that much of a problem?

Should we approach npm, inc, again about detailed download history for modules?

**Comment by ijroth** [comment]
How about a hybrid approach... Have slt-release add the postinstall script, which runs a module and passes it the name of the module whose postinstall is being run, which it then reports.

Also like the idea of using Google Analytics.

I'd like to put something like SL-blip into Express at some point but to do so we'd have to make the analytics available to the community in some form. So it's like a page on expressjs.com showing "look at all the places using express and look at all the installs going on, etc."

**Comment by rmg** [comment]
Unfortunately, the hybrid approach actually makes the shared dependency problem slightly more complicated because shared mid level dependencies don't get put into the $PATH (node-pre-gyp has this problem when it is not bundled). Reducing `scripts/post.js` to a single line that just requires a module would solve that problem, but would still require a script file. Another approach would be changing the postinstall script to something like `"node -e 'require(\"sl-blip\")'"`.

Note that npm sets the package name and version (and some other info) as environment variables when running scripts, so there's no need to update anything at release time or version bump.

The GA approach works as long as you can get the info you want from GA's reports. I'm not sure how to get the raw data out of GA for custom analysis. Another advantage to the GA approach is there is nothing proprietary (just pass your GA key as an argument) about it and that module could be open sourced and used by others, which could actually make it more acceptable to some people.

**Comment by rmg** [comment]
@sam-github the problem it is causing is the scary looking warnings and multiple retries that npm does to download the sl-blip URL when run on a server that doesn't have access to the real internet.

**Comment by ijroth** [comment]
Sounds like GA is the best solution then, for the reasons mentioned above. So we need to determine if we can get the type of information we want out of GA. Let's investigate with @sumitha

**Comment by rmg** [comment]
@ijroth a hybrid approach would be to use a postinstall to report to sl-blip, so that there's no change in the information being sent, just more control over if/when it is run.

File: package.json [hunk 1]
```diff
@@ -74,6 +74,7 @@
     "lb-ng-doc": "./node_modules/loopback-sdk-angular-cli/bin/lb-ng-doc"
   },
   "scripts": {
+    "postinstall": "node ./scripts/blip.min.js",
```
Comment (sam-github) [inline package.json]: could this be `sl-blip`, and sl-blip be the bin script of a strong-blip dependency?

**Comment by rmg** [comment]
Will be handled in strong-tools.

**Pull request closed without merging**